cmd/keymasterd/app.go

```diff
@@ -705,9 +703,7 @@ func (state *RuntimeState) checkAuth(w http.ResponseWriter, r *http.Request, req
 		state.Mutex.Lock()
 		config := state.Config
 		state.Mutex.Unlock()
-		if !state.Config.Base.DisableUsernameNormalization {
-			user = strings.ToLower(user)
-		}
+		user = state.reprocessUsername(user)
```

Comment:
these changes shuold not go into checkAuth. Only on the login handler (even better if on the web you add some JS to detects this and warns the user)

Reply:
In fact I am thinking that it should probably only live on the okta auth path...

Reply:
Adding JS to detect and warn the user would add friction for the user. I think we have to accept that they're already been trained to enter `username@company.com` and what's the value in beating them to split their habits?

As for just Okta: if another primary auth backend is added, it may have the same issue. Also, it seems to make sense to perform all these username cleanups in one place.